### PR TITLE
Make ssh key default to 0400 mode

### PR DESCRIPTION
**Pull request by krzyzacy**
WARNING: UNPROTECTED PRIVATE KEY FILE!

https://kubernetes.io/docs/user-guide/secrets/

### CODE REVIEW

**Comment by krzyzacy** [comment]
meh, not that easy :-( one sec

File: prow/cmd/line/main.go [hunk 1]
```diff
@@ -347,12 +347,14 @@ func (c *testClient) TestKubernetes() error {
 			Secret: &kube.SecretSource{
 				Name: "ssh-key-secret",
 			},
+			defaultMode: 256,
```
Comment (ixdy) [inline prow/cmd/line/main.go]: this isn't json.

Comment (krzyzacy) [inline prow/cmd/line/main.go]: https://github.com/kubernetes/test-infra/blob/master/prow/kube/client.go#L248

Comment (ixdy) [inline prow/cmd/line/main.go]: it's only an issue if it's specified in a json file. by the time this value gets there it'll be converted into its integer value.

File: prow/kube/types.go [hunk 2]
```diff
@@ -109,6 +109,8 @@ type Volume struct {
 	DownwardAPI *DownwardAPISource `json:"downwardAPI,omitempty"`
 	HostPath    *HostPathSource    `json:"hostPath,omitempty"`
 	ConfigMap   *ConfigMapSource   `json:"configMap,omitempty"`
+	// https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L954
```
Comment (spxtr) [inline prow/kube/types.go]: Linking to lines of code on master is always confusing. This one in particular is already wrong XD

**Comment by spxtr** [comment]
Why is this necessary?

**Comment by krzyzacy** [comment]
```
I0301 01:56:38.942] @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
I0301 01:56:38.942] @         WARNING: UNPROTECTED PRIVATE KEY FILE!          @
I0301 01:56:38.942] @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
I0301 01:56:38.942] Permissions 0644 for '/workspace/.ssh/google_compute_engine' are too open.
I0301 01:56:38.942] It is recommended that your private key files are NOT accessible by others.
I0301 01:56:38.942] This private key will be ignored.
I0301 01:56:38.943] key_load_private_type: bad permissions
I0301 01:56:38.943] Permission denied (publickey).
I0301 01:56:38.943] ERROR: (gcloud.compute.copy-files) [/usr/bin/scp] exited with return code [1].
I0301 01:56:38.943] See https://cloud.google.com/compute/docs/troubleshooting#ssherrors for troubleshooting hints.
```

File: prow/kube/types.go (hunk 2 above)
Comment (spxtr) [inline prow/kube/types.go]: I think you should just remove this comment.

File: prow/cmd/line/main.go [hunk 3]
```diff
@@ -347,12 +347,14 @@ func (c *testClient) TestKubernetes() error {
...
 				Name: "ssh-key-secret",
 			},
+			DefaultMode: 0400,
 		},
 		kube.Volume{
 			Name: "cache-ssd",
 			HostPath: &kube.HostPathSource{
 				Path: "/mnt/disks/ssd0",
 			},
+			DefaultMode: 0400,
```
Comment (spxtr) [inline prow/cmd/line/main.go]: The cache should not be 0400, no?

File: prow/kube/types.go [hunk 4]
```diff
@@ -109,6 +109,8 @@ type Volume struct {
...
+	DefaultMode int32 `json:"defaultMode,omitempty"`
```
Comment (ixdy) [inline prow/kube/types.go]: this is in the wrong object - `defaultMode` is part of `SecretVolumeSource` (`SecretSource` here), not `Volume`.

(this illustrates the need for the kubernetes client library. :)

Comment (krzyzacy) [inline prow/kube/types.go]: yeah you are right

File: prow/kube/types.go (hunk 2 above)
Comment (krzyzacy) [inline prow/kube/types.go]: yeah

File: prow/cmd/line/main.go (hunk 3 above)
Comment (krzyzacy) [inline prow/cmd/line/main.go]: oh sorry, wrong one :-p

Comment (krzyzacy) [inline prow/cmd/line/main.go]: I read `cache-ssh`

Comment (krzyzacy) [inline prow/cmd/line/main.go]: done.

File: prow/kube/types.go (hunk 4 above)
Comment (krzyzacy) [inline prow/kube/types.go]: done.

File: prow/Makefile [hunk 5]
```diff
@@ -16,7 +16,7 @@ all: build fmt vet test
 
 
 HOOK_VERSION       = 0.85
-LINE_VERSION       = 0.80
+LINE_VERSION       = 0.81
```
Comment (spxtr) [inline prow/Makefile]: 82 now

**Comment by spxtr** [comment]
LG after comment.

File: prow/Makefile (hunk 5 above)
Comment (krzyzacy) [inline prow/Makefile]: the world changes fast!

File: prow/cmd/line/main.go [hunk 6]
```diff
@@ -345,7 +345,8 @@ func (c *testClient) TestKubernetes() error {
 		kube.Volume{
 			Name: "ssh",
 			Secret: &kube.SecretSource{
-				Name: "ssh-key-secret",
+				Name:        "ssh-key-secret",
```
Comment (krzyzacy) [inline prow/cmd/line/main.go]: is my gofmt out of wreck... this indent does not look quite right

Comment (spxtr) [inline prow/cmd/line/main.go]: That's right. it lines up with the next line.

Comment (krzyzacy) [inline prow/cmd/line/main.go]: oh right :-p

**Pull request merged**